### PR TITLE
Road to implicitAny part 4

### PR DESCRIPTION
**Pull request by henrikra**
## What is this?
This PR continues the work on making this project type safe with TypeScript. Goal is that the flag `noImplicitAny` will be set to `true`

## What this PR does?
- all files inside `adapter` folder is valid for `noImplicitAny`!
- `resolveAssetSource` is now imported from the current place where it should. Previously it was imported from legacy location. More info here: https://facebook.github.io/react-native/docs/image#resolveassetsource
- formatted many files
- add missing interfaces
- fix `onPressIn` prop on `TouchablePreview`

## What's next?
Next I am planning to make same treatment for `commands` folder and many it valid for `noImplicitAny` 🎉

### CODE REVIEW

**Comment by henrikra** [comment]
@guyca Thx for merging my last PR so quickly! Here is next follow up :D

File: lib/src/adapters/NativeCommandsSender.ts [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+
 export class NativeCommandsSender {
-  private readonly nativeCommandsModule;
+  private readonly nativeCommandsModule: NativeCommandsModule;
   constructor() {
     this.nativeCommandsModule = NativeModules.RNNBridgeModule;
   }
 
-  setRoot(commandId: string, layout: { root: any, modals: any[], overlays: any[] }) {
+  setRoot(commandId: string, layout: { root: any; modals: any[]; overlays: any[] }) {
```
Comment (guyca) [inline lib/src/adapters/NativeCommandsSender.ts]: What is changing `,` to `;` needed?

File: lib/src/adapters/NativeEventsReceiver.ts [hunk 2]
```diff
@@ -7,22 +7,24 @@ import {
   SearchBarUpdatedEvent,
   SearchBarCancelPressedEvent,
   PreviewCompletedEvent,
-  ModalDismissedEvent
+  ModalDismissedEvent,
```
Comment (guyca) [inline lib/src/adapters/NativeEventsReceiver.ts]: Can we avoid adding comma character where it's avoidable?

File: lib/src/adapters/NativeEventsReceiver.ts [hunk 3]
```diff
@@ -31,39 +33,57 @@ export class NativeEventsReceiver {
     return this.emitter.addListener('RNN.AppLaunched', callback);
   }
 
-  public registerComponentDidAppearListener(callback: (event: ComponentDidAppearEvent) => void): EventSubscription {
+  public registerComponentDidAppearListener(
```
Comment (guyca) [inline lib/src/adapters/NativeEventsReceiver.ts]: Lets return these method declarations to single line - I think it's readable enough in a single line.

File: lib/src/adapters/NativeCommandsSender.ts (hunk 1 above)
Comment (henrikra) [reply lib/src/adapters/NativeCommandsSender.ts]: Yeah that is how types separate properties in TypeScript https://www.typescriptlang.org/docs/handbook/interfaces.html

**Comment by henrikra** [comment]
I have now done the changes you requested :)

**Pull request merged**